syncapi/storage/sqlite3/peeks_table.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlite3
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/matrix-org/dendrite/internal"
+	"github.com/matrix-org/dendrite/internal/sqlutil"
+	"github.com/matrix-org/dendrite/syncapi/storage/tables"
+	"github.com/matrix-org/dendrite/syncapi/types"
+)
+
+const peeksSchema = `
+CREATE TABLE IF NOT EXISTS syncapi_peeks (
+	id INTEGER PRIMARY KEY,
+	room_id TEXT NOT NULL,
+	user_id TEXT NOT NULL,
+	device_id TEXT NOT NULL,
+	new BOOL NOT NULL DEFAULT true,
+    -- When the peek was created in UNIX epoch ms.
+    creation_ts INTEGER NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS syncapi_peeks_room_id_idx ON syncapi_peeks(room_id);
+CREATE INDEX IF NOT EXISTS syncapi_peeks_user_id_device_id_idx ON syncapi_peeks(user_id, device_id);
+`
+
+const insertPeekSQL = "" +
+	"INSERT INTO syncapi_peeks" +
+	" (room_id, user_id, device_id, creation_ts)" +
+	" VALUES ($1, $2, $3, $4)"
+
+const deletePeekSQL = "" +
+	"DELETE FROM syncapi_peeks WHERE room_id = $1 AND user_id = $2 and device_id = $3"
+
+const deletePeeksSQL = "" +
+	"DELETE FROM syncapi_peeks WHERE room_id = $1 AND user_id = $2"
+
+const selectPeeksSQL = "" +
+	"SELECT room_id, new FROM syncapi_peeks WHERE user_id = $1 and device_id = $2"
+
+const selectPeekingDevicesSQL = "" +
+	"SELECT room_id, user_id, device_id FROM syncapi_peeks"
+
+const markPeeksAsOldSQL = "" +
+	"UPDATE syncapi_peeks SET new=false WHERE user_id = $1 and device_id = $2"
+
+type peekStatements struct {
+	db                            *sql.DB
+	streamIDStatements            *streamIDStatements
+	insertPeekStmt        		  *sql.Stmt
+	deletePeekStmt         		  *sql.Stmt
```

Comment:
Something went wildly wrong with the indentation here :D Better to use spaces rather than tabs.